Avoid segfault calling fd_close() when devpoll_state has not yet been
initialized by Comm::SelectLoopInit()

Reported, but not yet fixed, upstream:
https://bugs.squid-cache.org/show_bug.cgi?id=5314

Part of the fix for Oracle Solaris Bug 35949097

--- squid-6.9/src/comm/ModDevPoll.cc
+++ squid-6.9/src/comm/ModDevPoll.cc
@@ -169,6 +169,9 @@ */
 void
 Comm::SelectLoopInit(void)
 {
+    if (devpoll_state != NULL)
+        return;
+    
     /* allocate memory first before attempting to open poll device */
     /* This tracks the FD devpoll offset+state */
     devpoll_state = (struct _devpoll_state *)xcalloc(
@@ -224,6 +227,9 @@ debugs(5, 5, "FD " << fd << ", type=" <<
            ", handler=" << handler << ", client_data=" << client_data <<
            ", timeout=" << timeout);
 
+    if (devpoll_state == NULL)
+        SelectLoopInit();
+
     /* POLLIN/POLLOUT are defined in <sys/poll.h> */
     fde *F = &fd_table[fd];
     if (!F->flags.open) {
